apps/prairielearn/src/pages/partials/navbarInstructor.ejs

```diff
@@ -1,23 +1,37 @@
 
     <%# Course admin %> 
-    <li class="nav-item btn-group">
+    <li class="nav-item btn-group" id="navbar-course-switcher">
         <a class="nav-link" <% if (navPage == "course_admin" && !(typeof navSubPage !== 'undefined' && (navSubPage == "issues" || navSubPage == "questions" || navSubPage == "syncs"))) { %>active<% } %> <% if (! authz_data.has_course_permission_view) { %> disabled <% } %>"
          href="<%= urlPrefix %>/course_admin" role="button">
             <%= course.short_name %>
         </a>
-        <% if ((authz_data.has_course_permission_preview || authz_data.has_course_instance_permission_view) && authz_data.authn_courses && authz_data.authn_courses.length > 0) { %>
-        <a class="nav-link dropdown-toggle dropdown-toggle-split" href="#" id="navbarDropdownMenuCourseAdminLink" role="button" data-toggle="dropdown" aria-label="Change course" aria-haspopup="true" aria-expanded="false">
+        <a
+            class="nav-link dropdown-toggle dropdown-toggle-split"
+            href="#"
+            id="navbarDropdownMenuCourseAdminLink"
+            role="button"
+            data-toggle="dropdown"
+            aria-label="Change course"
+            aria-haspopup="true"
+            aria-expanded="false"
+            <% if (!authz_data.overrides) { %>
+            hx-get="/pl/navbar/course/<%= course.id %>/switcher"
+            hx-trigger="show-course-switcher once delay:200ms"
+            hx-target="#navbarDropdownMenuCourseAdmin"
+            <% } %>
+        >
         </a>
         <div class="dropdown-menu" aria-labelledby="navbarDropdownMenuCourseAdminLink" id="navbarDropdownMenuCourseAdmin">
             <% if (authz_data.overrides) { %>
                 <span class="dropdown-item-text small">Effective users may not switch between courses</span>
             <% } else { %>
-                <% authz_data.authn_courses.forEach(function(c) { %>
-                    <a class="dropdown-item <% if (course.id == c.id) { %>active<% } %>" href="<%= plainUrlPrefix %>/course/<%= c.id %>/course_admin<% if (navPage == "course_admin" && typeof navSubPage !== 'undefined' && navSubPage != "file_edit") { %>/<%= navSubPage %><% } %>"><%= c.short_name %></a>
```

Comment:
I specifically chose to remove the behavior that would try to retain the current subpage when navigating between course instances, as I felt it would be a little too tricky to implement with the dynamic loading and I'm also not sure there's actually a ton of value to it. That said, it's possible to implement, probably by tacking on query parameters to the `hx-get` attribute that can reflect `navPage` and `navSubPage`.

Reply:
I don't see this as deal breaker, but it is sometimes nice to navigate, e.g., between issues pages of multiple courses, or between gradebooks of different instances. So I don't mind that much if this is removed, but it would be a nice to have feature to keep if it's not a big ask.

Reply:
I'm fine to remove this behavior. We've been thinking about a major refresh of the nav interface anyway, which would most likely make this behavior unnecessary.